server/src/main/java/org/elasticsearch/script/ScoreScript.java

```diff
@@ -104,10 +112,52 @@ public void setScorer(Scorable scorer) {
         };
     }
 
+    /**
+     * Accessed as _score in the painless script
+     * @return the score of the inner query
+     */
     public double get_score() {
         return scoreSupplier.getAsDouble();
     }
 
+    /**
+     * @return the internal document ID
+     */
+    public int getDocId() {
```

Comment:
I'm wondering if we should rename these methods to something like _getDocId().  This way the user cannot access them directly through a script.

Reply:
@jdconrad Thanks Jack, I will rename these method